apps/etterlatte-trygdetid/src/test/kotlin/trygdetid/avtale/AvtaleRepositoryTest.kt

```diff
@@ -0,0 +1,98 @@
+package no.nav.etterlatte.trygdetid
+
+import io.kotest.matchers.shouldBe
+import io.mockk.every
+import io.mockk.mockk
+import no.nav.etterlatte.libs.common.behandling.DetaljertBehandling
+import no.nav.etterlatte.libs.database.DataSourceBuilder
+import no.nav.etterlatte.libs.database.migrate
+import no.nav.etterlatte.trygdetid.avtale.AvtaleRepository
+import org.junit.jupiter.api.AfterAll
+import org.junit.jupiter.api.AfterEach
+import org.junit.jupiter.api.BeforeAll
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.TestInstance
+import org.testcontainers.containers.PostgreSQLContainer
+import org.testcontainers.junit.jupiter.Container
+import trygdetid.trygdeavtale
+import java.util.*
+import java.util.UUID.randomUUID
+import javax.sql.DataSource
+
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+internal class AvtaleRepositoryTest {
+
+    @Container
+    private val postgres = PostgreSQLContainer<Nothing>("postgres:14")
+    private lateinit var repository: AvtaleRepository
+    private lateinit var dataSource: DataSource
+
+    @BeforeAll
+    fun beforeAll() {
+        postgres.start()
+        dataSource = DataSourceBuilder.createDataSource(postgres.jdbcUrl, postgres.username, postgres.password)
+        repository = AvtaleRepository(dataSource.apply { migrate() })
+    }
+
+    @AfterAll
+    fun afterAll() {
+        postgres.stop()
+    }
+
+    @AfterEach
+    fun afterEach() {
+        cleanDatabase()
+    }
+
+    @Test
+    fun `skal opprette og hente avtaler`() {
+        val behandling = behandlingMock()
+        val avtale = trygdeavtale(behandling.id, "EOS_NOR", "EOS2010", "YRK_MEDL")
+
+        repository.opprettAvtale(avtale)
+
+        with(repository.hentAvtale(behandling.id)) {
+            this?.avtaleKode shouldBe avtale.avtaleKode
+            this?.avtaleDatoKode shouldBe avtale.avtaleDatoKode
+            this?.avtaleKriteriaKode shouldBe avtale.avtaleKriteriaKode
+            this?.id shouldBe avtale.id
+            this?.behandlingId shouldBe behandling.id
+            this?.kilde?.ident shouldBe "Z123"
+        }
```

Comment:
Siden `Trygdeavtale` er `data class` kan du bare bruke innebygget equals. Sparer en for mye unødvendig kode. 

```
val faktiskAvtale = repository.hentAvtale(behandling.id

faktiskAvtale shouldBe avtale
```

Samme i testen under.

Reply:
Jeg ville få frem at behandlingId ikke endret seg ved å sammenligne det med behandling.id - men kan forenkle